backend/day-15/day-15-challenge.py

```diff
@@ -0,0 +1,144 @@
+            
+            for website, account in data.items():
+                print(f"Website: {self.getWebsiteName(website)}")
+                for details in account:
+                    print(f"\tEmail: {details['email']}")
+                    print(f"\tPassword: {details['password']}")
+        except:
+            print("An error occured while fetching the data!")
+
+    def deleteData(self):
+        try:
+            data = self.__loadData('backend/day-15/data.json')
+            website_key = self.getWebsiteKey()
+            for account in data.get(website_key, []):
+                if account.get('email') == self.email and account.get('password') == self.password:
+                    data[website_key].remove(account)
+                    self.__updateData('backend/day-15/data.json', data)
+                    print("Password deleted successfully!")
+                    return
+            print("Email and password combination not found in the record! Please check your inputs and try again")
+        except FileNotFoundError:
+            print("File not found! Please check the file path")
+        except Exception as e:
+            print(f"An error occurred: {e}")
+
+    def updateData(self):
+        print("Update data function")
+
+    def getWebsiteName(self, code: str):
+        try:
+            data = self.__loadData('backend/day-15/websites.json')
+
+            for website in data:
+                if code == website['code']:
+                    return website['name']
+        except:
+            print("Error fetching website names!")
+    
+    def getWebsiteKey(self):
+        try:
+            data = self.__loadData('backend/day-15/websites.json')
+
+            for website in data:
+                if self.website.lower() == website['name'].lower():
+                    return website['code']
+
+            return self.website.lower().replace(' ', '_')    
+        except:
+            print("Error fetching website keys")
+    
+def generateDivider():
+    i = 50
+    counter = 1
+    while not counter > i:
+        print('*', end='')
+        counter += 1
+    print()
+
+def start():
+    options = (
+        (1, "Add a password"),
+        (2, "View passwords"),
+        (3, "Search"),
+        (4, "Delete a password"),
+        (5, "Update a password") 
+    )
+    
+    while True:
+        password_manager = PasswordManager("", "", "")
+
+        print("OPTIONS")
+        for process_number, process_description in options:
+            print(f"  {process_number} - {process_description}")
+        
+        try:
+            selected_option = options[(int(input("Select an option: "))) - 1][1].split()[0].upper()
+
+            generateDivider();
+            if selected_option == "ADD":
+                website = input("Website name: ")
+                email = input("Email: ")
+                password = input("Password: ")
+                PasswordManager(website, email, password).addData()
```

Comment:
Correct me if I'm wrong but I think `PasswordManager.addData(website, email, password)` makes more sense than `PasswordManager(website, email, password).addData()`

Is your approach a standard in python?

Reply:
I believe it's the `PasswordManager(website, email, password).addData()` just so I could access the passed data all throughout the `PasswordManager` class without the need of passing parameters. For instance, if I have to generate all passwords using the input website to the `getWebsiteKey()`, using the first option, I'll still have to pass the input website to it.

Mas robust sya kumbaga.

"Is your approach a standard in python?"
HAHAHAHAJHAHAH ewan